andela/octopus-bn-backend
# #170947437 Build password reset feature

### What does this PR do?

- Create Forgot password endpoint
- Create reset password endpoint
- Remove useless comments
- Write tests for reset password feature
- add validation for newPassword field

### How should this be manually tested?

- Clone the [repo](https://github.com/andela/octopus-bn-backend.git)
- Run ``` npm install ```
-  Hit 
  -  `POST ``http://localhost:3000/api/v1/auth/forgot-password` to request a password reset link.
  - Check email for the confirmation link and run it using a `POST` method

### Implementation plan

[Implementation plan name](https://docs.google.com/document/d/11UF7-_F-_vS5aZ0oAmVbuYwDeEZfZS_sHjq6U-h0vXQ/edit)

### Stories associated with PR

[##170947435](https://www.pivotaltracker.com/story/show/170947437)

### Screenshots if any
<img width="1175" alt="Screen Shot 2020-02-18 at 20 11 16" src="https://user-images.githubusercontent.com/54870069/74764701-dd4d0c00-528a-11ea-9cdd-b677df2d5a6d.png">
<img width="1182" alt="Screen Shot 2020-02-18 at 20 11 53" src="https://user-images.githubusercontent.com/54870069/74764758-f950ad80-528a-11ea-80b1-d3743a91f69a.png">
